src/main/java/org/dita/dost/writer/ProfilingFilter.java

```diff
@@ -68,31 +72,30 @@ public boolean hasElementOutput() {
 	@Override
 	public void startElement(final String uri, final String localName, final String qName, final Attributes atts)
 			throws SAXException {
-		if (foreignLevel > 0) {
-			foreignLevel++;
-		} else if (foreignLevel == 0) {
-			final String classAttr = atts.getValue(ATTRIBUTE_NAME_CLASS);
-			if (classAttr == null && !ELEMENT_NAME_DITA.equals(localName)) {
-				logger.info(MessageUtils.getInstance().getMessage("DOTJ030I", localName).toString());
-			}
-			if (classAttr != null && (TOPIC_TOPIC.matches(classAttr) || MAP_MAP.matches(classAttr))) {
-				final String domains = atts.getValue(ATTRIBUTE_NAME_DOMAINS);
-				if (domains == null) {
-					logger.info(MessageUtils.getInstance().getMessage("DOTJ029I", localName).toString());
-				} else {
-					props = StringUtils.getExtProps(domains);
-				}
-			}
-			if (classAttr != null && (TOPIC_FOREIGN.matches(classAttr) || TOPIC_UNKNOWN.matches(classAttr))) {
-				foreignLevel = 1;
+		final DitaClass cls = atts.getValue(ATTRIBUTE_NAME_CLASS) != null ? new DitaClass(atts.getValue(ATTRIBUTE_NAME_CLASS)) : null;
+        classes.addFirst(cls);
+        /*if ((cls==null || !cls.isValidDitaClass()) && !ELEMENT_NAME_DITA.equals(localName)) {
+        	if (nonDitaContext(classes)) {
+        		// Normal case for bad class: in non DITA context, no message
+        	} else if (cls==null) {
+        		logger.info(MessageUtils.getInstance().getMessage("DOTJ030I", localName).setLocation(atts).toString());
+        	} else { // !cls.isValidDitaClass()
+        		logger.info(MessageUtils.getInstance().getMessage("DOTJ070I", atts.getValue(ATTRIBUTE_NAME_CLASS), localName).setLocation(atts).toString());
+        	}
+        } else */ if (cls != null && (TOPIC_TOPIC.matches(cls) || MAP_MAP.matches(cls))) {
+			final String domains = atts.getValue(ATTRIBUTE_NAME_DOMAINS);
+			if (domains == null) {
+				logger.info(MessageUtils.getInstance().getMessage("DOTJ029I", localName).toString());
+			} else {
+				props = StringUtils.getExtProps(domains);
 			}
 		}
 
 		if (exclude) {
 			// If it is the start of a child of an excluded tag, level increase
 			level++;
 		} else { // exclude shows whether it's excluded by filtering
-			if (foreignLevel <= 1 && filterUtils.needExclude(atts, props)) {
+			if (cls!=null && filterUtils.needExclude(atts, props)) {
```

Comment:
In this case -- changing from `cls!=null` to `cls.isValidDitaClass()` is what we need to prevent filtering non-DITA content inside the foreign / unknown context.